hypervisor/network.go

```diff
@@ -282,13 +284,61 @@ func (nc *NetworkContext) getInterface(id string) *InterfaceCreated {
 	return nil
 }
 
+func (nc *NetworkContext) AddIPAddr(id, ip string) error {
+	nc.slotLock.RLock()
+	defer nc.slotLock.RUnlock()
+	inf, ok := nc.idMap[id]
+	if !ok {
+		return ErrNoSuchInf
+	}
+	inf.IpAddr = append(inf.IpAddr, ip)
+	inf.Desc.Ip = append(inf.Desc.Ip, ip)
+	return nil
+}
+
+func (nc *NetworkContext) DeleteIPAddr(id, ip string) error {
+	nc.slotLock.RLock()
+	defer nc.slotLock.RUnlock()
+	inf, ok := nc.idMap[id]
+	if !ok {
+		return ErrNoSuchInf
+	}
+
+	for i, v := range inf.IpAddr {
+		if v == ip {
+			inf.IpAddr = append(inf.IpAddr[:i], inf.IpAddr[i+1:]...)
```

Comment:
what happen when the last element in ipaddr array match the ip, is the inf.IpAddr[i+1] will access the array out of range?